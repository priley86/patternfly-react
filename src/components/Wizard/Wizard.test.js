import React from 'react';
import { mount } from 'enzyme';
import { Button, Row, Col, Wizard } from '../../index';

test('Wizard loading renders properly', () => {
  const component = mount(
    <Row>
      <Col sm={12}>
        <Wizard>
          <Wizard.Header title="Wizard Title" />
          <Wizard.Body>
            <Wizard.Row>
              <Wizard.Main>Contents</Wizard.Main>
            </Wizard.Row>
          </Wizard.Body>
          <Wizard.Footer>
            <Button bsStyle="default" className="btn-cancel">
              Cancel
            </Button>
            <Button bsStyle="default" disabled>
              <span className="i fa fa-angle-left" />Back
            </Button>
            <Button bsStyle="primary" disabled>
              Next<span className="i fa fa-angle-right" />
            </Button>
          </Wizard.Footer>
        </Wizard>
      </Col>
    </Row>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard steps renders properly', () => {
  const onStepClick = jest.fn();

  const component = mount(
    <Wizard>
      <Wizard.Header title="Wizard Title" />
      <Wizard.Body>
        <Wizard.Steps
          steps={[
            <Wizard.Step
              key={0}
              stepIndex={0}
              step={0}
              label="Step 1"
              title="Step 1"
              onClick={onStepClick}
            >
              <Wizard.SubStep key={0} subStep="1.1" title="Step 1.1" />
            </Wizard.Step>
          ]}
        />
      </Wizard.Body>
    </Wizard>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard sidebar renders properly', () => {
  const onSidebarItemClick = jest.fn();
  const component = mount(
    <Wizard>
      <Wizard.Header title="Wizard Title" />
      <Wizard.Body>
        <Wizard.Row>
          <Wizard.Sidebar
            items={[
              <Wizard.SidebarGroup key="1" step="1" activeStep="1">
                <Wizard.SidebarGroupItem
                  key="1.1"
                  stepIndex={1}
                  subStepIndex={1}
                  subStep="1.1"
                  label="1A."
                  title="Details"
                  activeSubStep="1.1"
                  onClick={onSidebarItemClick}
                />
              </Wizard.SidebarGroup>
            ]}
          />
        </Wizard.Row>
      </Wizard.Body>
    </Wizard>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard main contents renders properly', () => {
  const component = mount(
    <Wizard>
      <Wizard.Header title="Wizard Title" />
      <Wizard.Body>
        <Wizard.Row>
          <Wizard.Main>
            <Wizard.Contents
              key={0}
              stepIndex={1}
              subStepIndex={1}
              activeStepIndex={1}
              activeSubStepIndex={1}
            >
              Content
            </Wizard.Contents>
          </Wizard.Main>
        </Wizard.Row>
      </Wizard.Body>
    </Wizard>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard review contents renders properly', () => {
  const onStepClicked = jest.fn();
  const onSubStepClicked = jest.fn();
  const component = mount(
    <Wizard>
      <Wizard.Header title="Wizard Title" />
      <Wizard.Body>
        <Wizard.Row>
          <Wizard.Main>
            <Wizard.ReviewSteps>
              <Wizard.ReviewStep
                title="Step 1 Review"
                collapsed
                onClick={onStepClicked}
              >
                <Wizard.ReviewSubSteps collapsed>
                  <Wizard.ReviewSubStep
                    label="Review 1.1"
                    title="Review 1.1"
                    collapsed
                    onClick={onSubStepClicked}
                  >
                    <Wizard.ReviewContent collapsed>
                      <Wizard.ReviewItem>
                        <span className="wizard-pf-review-item-label">
                          {'Step 1.1 Label'}:
                        </span>
                        <span className="wizard-pf-review-item-value">
                          Brian Johnson
                        </span>
                      </Wizard.ReviewItem>
                      <Wizard.ReviewItem>
                        <span className="wizard-pf-review-item-label">
                          {'Step 1.2 Label'}:
                        </span>
                        <span className="wizard-pf-review-item-value">
                          This is the description.
                        </span>
                      </Wizard.ReviewItem>
                    </Wizard.ReviewContent>
                  </Wizard.ReviewSubStep>
                </Wizard.ReviewSubSteps>
              </Wizard.ReviewStep>
            </Wizard.ReviewSteps>
          </Wizard.Main>
        </Wizard.Row>
      </Wizard.Body>
    </Wizard>
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard Pattern renders properly', () => {
  const onHide = jest.fn();
  const onExited = jest.fn();
  const onStepChanged = jest.fn();
  const component = mount(
    <Wizard.Pattern
      show
      onHide={onHide}
      onExited={onExited}
      title="Wizard Pattern Example"
      shouldDisableNextStep={false}
      steps={[{ title: 'General', render: () => <p>General</p> }]}
      loadingTitle="Loading..."
      loadingMessage="This may take a minute."
      loading
      activeStepIndex={0}
      onStepChanged={onStepChanged}
    />
  );

  expect(component.render()).toMatchSnapshot();
});

test('Wizard Stateful Pattern renders properly', () => {
  const onHide = jest.fn();
  const onExited = jest.fn();
  const component = mount(
    <Wizard.Pattern.Stateful
      show
      activeStepIndex={0}
      onHide={onHide}
      onExited={onExited}
      title="Wizard Pattern Stateful Example"
      shouldDisableNextStep={() => false}
      shouldPreventStepChange={() => false}
      steps={[{ title: 'General', render: () => <p>General</p> }]}
      loadingTitle="Loading..."
      loadingMessage="This may take a minute."
      loading
    />
  );

  expect(component.render()).toMatchSnapshot();
});
