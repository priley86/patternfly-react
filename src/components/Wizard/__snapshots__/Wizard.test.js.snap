// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard Pattern renders properly 1`] = `
<div
  role="dialog"
>
  <div
    class="modal-backdrop fade in"
  />
  <div
    class="fade in modal"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-lg wizard-pf modal-dialog"
    >
      <div
        class="modal-content"
        role="document"
      >
        <div
          class=""
        >
          <div
            class="modal-header"
          >
            <button
              aria-hidden="true"
              aria-label="Close"
              class="close"
            >
              <span
                aria-hidden="true"
                class="pficon pficon-close"
              />
            </button>
            <h4
              class="modal-title"
            >
              Wizard Pattern Example
            </h4>
          </div>
          <div
            class="wizard-pf-body clearfix modal-body"
          >
            <section
              class="wizard-pf-row"
            >
              <div
                class="wizard-pf-main"
              >
                <div
                  class="blank-slate-pf"
                >
                  <div
                    class="blank-slate-pf-icon spinner spinner-lg"
                  />
                  <div
                    class="blank-slate-pf-main-action"
                  >
                    <h3>
                      Loading...
                    </h3>
                  </div>
                  <div
                    class="blank-slate-pf-secondary-action"
                  >
                    <p>
                      This may take a minute.
                    </p>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="wizard-pf-footer modal-footer"
          >
            <button
              class="btn-cancel btn btn-default"
              type="button"
            >
              Cancel
            </button>
            <button
              class="btn btn-default"
              disabled=""
              type="button"
            >
              <span
                aria-hidden="true"
                class="fa fa-angle-left"
              />
              Back
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wizard Stateful Pattern renders properly 1`] = `
<div
  role="dialog"
>
  <div
    class="modal-backdrop fade in"
  />
  <div
    class="fade in modal"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-lg wizard-pf modal-dialog"
    >
      <div
        class="modal-content"
        role="document"
      >
        <div
          class=""
        >
          <div
            class="modal-header"
          >
            <button
              aria-hidden="true"
              aria-label="Close"
              class="close"
            >
              <span
                aria-hidden="true"
                class="pficon pficon-close"
              />
            </button>
            <h4
              class="modal-title"
            >
              Wizard Pattern Stateful Example
            </h4>
          </div>
          <div
            class="wizard-pf-body clearfix modal-body"
          >
            <section
              class="wizard-pf-row"
            >
              <div
                class="wizard-pf-main"
              >
                <div
                  class="blank-slate-pf"
                >
                  <div
                    class="blank-slate-pf-icon spinner spinner-lg"
                  />
                  <div
                    class="blank-slate-pf-main-action"
                  >
                    <h3>
                      Loading...
                    </h3>
                  </div>
                  <div
                    class="blank-slate-pf-secondary-action"
                  >
                    <p>
                      This may take a minute.
                    </p>
                  </div>
                </div>
              </div>
            </section>
          </div>
          <div
            class="wizard-pf-footer modal-footer"
          >
            <button
              class="btn-cancel btn btn-default"
              type="button"
            >
              Cancel
            </button>
            <button
              class="btn btn-default"
              disabled=""
              type="button"
            >
              <span
                aria-hidden="true"
                class="fa fa-angle-left"
              />
              Back
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wizard loading renders properly 1`] = `
<div
  class="row"
>
  <div
    class="col-sm-12"
  >
    <div
      class=""
    >
      <div
        class="wizard-pf-header"
      >
        <h4
          class="wizard-pf-title"
        >
          Wizard Title
        </h4>
      </div>
      <div
        class="wizard-pf-body"
      >
        <section
          class="wizard-pf-row"
        >
          <div
            class="wizard-pf-main"
          >
            Contents
          </div>
        </section>
      </div>
      <div
        class="wizard-pf-footer"
      >
        <button
          class="btn-cancel btn btn-default"
          type="button"
        >
          Cancel
        </button>
        <button
          class="btn btn-default"
          disabled=""
          type="button"
        >
          <span
            class="i fa fa-angle-left"
          />
          Back
        </button>
        <button
          class="btn btn-primary"
          disabled=""
          type="button"
        >
          Next
          <span
            class="i fa fa-angle-right"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wizard main contents renders properly 1`] = `
<div
  class=""
>
  <div
    class="wizard-pf-header"
  >
    <h4
      class="wizard-pf-title"
    >
      Wizard Title
    </h4>
  </div>
  <div
    class="wizard-pf-body"
  >
    <section
      class="wizard-pf-row"
    >
      <div
        class="wizard-pf-main"
      >
        <div
          class="wizard-pf-contents"
        >
          Content
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Wizard review contents renders properly 1`] = `
<div
  class=""
>
  <div
    class="wizard-pf-header"
  >
    <h4
      class="wizard-pf-title"
    >
      Wizard Title
    </h4>
  </div>
  <div
    class="wizard-pf-body"
  >
    <section
      class="wizard-pf-row"
    >
      <div
        class="wizard-pf-main"
      >
        <div
          class="wizard-pf-review-steps"
        >
          <ul
            class="list-group"
          >
            <li
              class="list-group-item"
            >
              <a
                class="collapsed"
                href="#"
              >
                Step 1 Review
              </a>
              <div
                class="wizard-pf-review-substeps collapse"
              >
                <ul
                  class="list-group"
                >
                  <li
                    class="list-group-item"
                  >
                    <a
                      class="collapsed"
                      href="#"
                    >
                      <span
                        class="wizard-pf-substep-number"
                      >
                        Review 1.1
                      </span>
                      <span
                        class="wizard-pf-substep-title"
                      >
                        Review 1.1
                      </span>
                    </a>
                    <div
                      class="wizard-pf-review-content collapse"
                    >
                      <div
                        class="wizard-pf-review-item"
                      >
                        <span
                          class="wizard-pf-review-item-label"
                        >
                          Step 1.1 Label:
                        </span>
                        <span
                          class="wizard-pf-review-item-value"
                        >
                          Brian Johnson
                        </span>
                      </div>
                      <div
                        class="wizard-pf-review-item"
                      >
                        <span
                          class="wizard-pf-review-item-label"
                        >
                          Step 1.2 Label:
                        </span>
                        <span
                          class="wizard-pf-review-item-value"
                        >
                          This is the description.
                        </span>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`Wizard sidebar renders properly 1`] = `
<div
  class=""
>
  <div
    class="wizard-pf-header"
  >
    <h4
      class="wizard-pf-title"
    >
      Wizard Title
    </h4>
  </div>
  <div
    class="wizard-pf-body"
  >
    <section
      class="wizard-pf-row"
    >
      <div
        class="wizard-pf-sidebar"
      >
        <ul
          class="list-group"
        >
          <li
            class="active list-group-item"
          >
            <a
              href="#"
            >
              <span
                class="wizard-pf-substep-number"
              >
                1A.
              </span>
              <span
                class="wizard-pf-substep-title"
              >
                Details
              </span>
            </a>
          </li>
        </ul>
      </div>
    </section>
  </div>
</div>
`;

exports[`Wizard steps renders properly 1`] = `
<div
  class=""
>
  <div
    class="wizard-pf-header"
  >
    <h4
      class="wizard-pf-title"
    >
      Wizard Title
    </h4>
  </div>
  <div
    class="wizard-pf-body"
  >
    <div
      class="wizard-pf-steps"
    >
      <ul
        class="wizard-pf-steps-indicator"
      >
        <li
          class="wizard-pf-step"
        >
          <a
            href="#"
          >
            <span
              class="wizard-pf-step-number"
            >
              Step 1
            </span>
            <span
              class="wizard-pf-step-title"
            >
              Step 1
            </span>
            <span
              class="wizard-pf-step-title-substep"
            >
              Step 1.1
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
